softlayer/vm/softlayer_virtual_guest_deleter.go

```diff
@@ -40,6 +42,16 @@ func (c *softLayerVMDeleter) Delete(cid int) error {
 		}
 	}
 
+	vm, found, err := c.vmFinder.Find(cid)
+	if err != nil || !found {
+		return bosherr.WrapErrorf(err, "Cannot find VirtualGuest with id: %d.", cid)
```

Comment:
Can we separate these two cases? I.e.,

1. when there is `err`
2. when `!found`

I think that will make code easier to read and most importantly to test.